syrillian/src/engine/rendering/proxies/skybox_proxy.rs

```diff
@@ -0,0 +1,229 @@
+        });
+
+        // Create cubemap bind group if texture is available
+        let cubemap_bind_group = if let Some(cubemap_texture) = renderer.cache.cubemap(self.cubemap)
+        {
+            let layout = device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+                label: Some("Skybox Bind Group Layout"),
+                entries: &[
+                    wgpu::BindGroupLayoutEntry {
+                        binding: 0,
+                        visibility: wgpu::ShaderStages::FRAGMENT,
+                        ty: wgpu::BindingType::Texture {
+                            multisampled: false,
+                            view_dimension: wgpu::TextureViewDimension::Cube,
+                            sample_type: wgpu::TextureSampleType::Float { filterable: true },
+                        },
+                        count: None,
+                    },
+                    wgpu::BindGroupLayoutEntry {
+                        binding: 1,
+                        visibility: wgpu::ShaderStages::FRAGMENT,
+                        ty: wgpu::BindingType::Sampler(wgpu::SamplerBindingType::Filtering),
+                        count: None,
+                    },
+                ],
+            });
+
+            let sampler = device.create_sampler(&wgpu::SamplerDescriptor {
+                label: Some("Skybox Sampler"),
+                address_mode_u: wgpu::AddressMode::ClampToEdge,
+                address_mode_v: wgpu::AddressMode::ClampToEdge,
+                address_mode_w: wgpu::AddressMode::ClampToEdge,
+                mag_filter: wgpu::FilterMode::Linear,
+                min_filter: wgpu::FilterMode::Linear,
+                mipmap_filter: wgpu::FilterMode::Linear,
+                ..Default::default()
+            });
+
+            Some(device.create_bind_group(&wgpu::BindGroupDescriptor {
+                label: Some("Skybox Bind Group"),
+                layout: &layout,
+                entries: &[
+                    wgpu::BindGroupEntry {
+                        binding: 0,
+                        resource: wgpu::BindingResource::TextureView(&cubemap_texture.view),
+                    },
+                    wgpu::BindGroupEntry {
+                        binding: 1,
+                        resource: wgpu::BindingResource::Sampler(&sampler),
+                    },
+                ],
+            }))
+        } else {
+            None
+        };
+
+        Box::new(RuntimeSkyboxData {
+            cubemap_bind_group,
+            uniform,
+            vertex_buffer,
+            vertex_count: SKYBOX_VERTICES.len() as u32,
+        })
+    }
+
+    fn update_render(
+        &mut self,
+        _renderer: &Renderer,
+        data: &mut dyn Any,
+        _window: &Window,
+        _local_to_world: &Matrix4<f32>,
+    ) {
+        let _data: &mut RuntimeSkyboxData = proxy_data_mut!(data);
+        // Camera uniform updates are handled automatically by the renderer
```

Comment:
That comment is pointless then. Also, you haven't handled any updates to the skybox proxy...